File: src/main/java/de/nordgedanken/auto_hot_key/psi/AHKFile.java
package de.nordgedanken.auto_hot_key.psi;

import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import de.nordgedanken.auto_hot_key.AHKFileType;
import de.nordgedanken.auto_hot_key.AHKLanguage;
import org.jetbrains.annotations.NotNull;

public class AHKFile extends PsiFileBase {
    public AHKFile(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, AHKLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public FileType getFileType() {
        return AHKFileType.INSTANCE;
    }

    @Override
    public String toString() {
        return "AutoHotKey Script";
    }
}

File: src/main/kotlin/com/autohotkey/lang/psi/AhkFile.kt
package com.autohotkey.lang.psi

import com.autohotkey.lang.core.AhkFileType
import com.autohotkey.lang.core.AhkLanguage
import com.intellij.extapi.psi.PsiFileBase
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider

class AhkFile(viewProvider: FileViewProvider) : PsiFileBase(viewProvider, AhkLanguage) {
    override fun getFileType(): FileType = AhkFileType

    override fun toString() = "AutoHotkey Script File"
}
